src/libsystemd-network/ndisc-router.c

```diff
@@ -530,10 +530,10 @@ int sd_ndisc_router_route_get_preference(sd_ndisc_router *rt, unsigned *ret) {
         if (r < 0)
                 return r;
 
-        *ret = (ri[3] >> 3) & 3;
-        if (!IN_SET(*ret, SD_NDISC_PREFERENCE_LOW, SD_NDISC_PREFERENCE_HIGH))
-                *ret = SD_NDISC_PREFERENCE_MEDIUM;
+        if (!IN_SET((ri[3] >> 3) & 3, SD_NDISC_PREFERENCE_LOW, SD_NDISC_PREFERENCE_HIGH))
```

Comment:
How about to introduce tentative variable to store `(ri[3] >> 3) & 3` ??
Missing `SD_NDISC_PREFERNCE_MEDIUM` in `IN_SET()`.

Reply:
It's OK as a bit operation

Reply:
At least, `SD_NDISC_PREFERNCE_MEDIUM` needs to be added in `IN_SET()`.

Reply:
yes updated thanks !